src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -1001,6 +1001,43 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                                 //[nameof(Constants.Fields.CommandData.DocumentChangeVector)] = tsCmd.LastDocumentChangeVector
                             });
 
+                            break;
+                        case CommandType.TimeSeriesCopy:
+
+                            var names = Database.DocumentsStorage.TimeSeriesStorage.GetTimeSeriesNamesForDocument(context, cmd.Id);
+                            foreach (var name in names)
+                            {
+                                var reader = Database.DocumentsStorage.TimeSeriesStorage.GetReader(context, cmd.Id, name.ToString(), DateTime.MinValue, DateTime.MaxValue);
+
+                                var tso = new TimeSeriesOperation()
+                                {
+                                    Name = name.ToString()
+                                };
+
+                                foreach (var singleResult in reader.AllValues())
+                                {
+                                    var temp = new TimeSeriesOperation.AppendOperation()
+                                    {
+                                        Tag = singleResult.Tag,
+                                        Timestamp = singleResult.Timestamp,
+                                        Values = singleResult.Values.ToArray()
+                                    };
+                                    tso.Append(temp);
```

Comment:
Instead of materialize all of the results at once, we can directly append them.
should be something like `AppendTimestamp(context, cmd.Id, collection, name, reader.AllValues())`